### ISSUE TITLE
A segmentation issue when using large packet size with Time Aware Shaper

### ISSUE THREAD

**Issue by howhangliu** [inet-framework/inet#895]
Hello Levy,

I have been experimenting with the TAS feature using various configurations and have identified an issue that I would like to report to you.

In the INI file **[Scenario_3]**, I use a `UdpSourceApp` and set the `packetLength` to a large number. i.e., 6300B.

Then there is an error message pops up as follows:

_A runtime error occurred:
omnetpp::common::expression::ExprNode::eval_error: Object nullptr has no member named 'destPort' -- in module (inet::queueing::PacketMultiplexer) cutthroughTAS.tactile.bridging.directionReverser.join (id=301), at t=0s, event 11_

I have located the source of the issue to be in PacketFilter.cc on line 109. 
However, I am unsure of how to resolve it. 

Would it be possible for you to provide me with some guidance on this matter? 
Thank you for your assistance.

Here are the ini and ned files.
[ini_ned.zip](https://github.com/inet-framework/inet/files/11790216/ini_ned.zip)

**Comment by levy** [comment]
This issue is caused by the IP fragmentation. The packet length is large enough for the IP layer to fragment the datagram.

If a packet is fragmented in the IP layer, then the packet dissector refuses to dissect the IP service data unit (SDU). This happens because the IP protocol field cannot be used reliably for dissecting the fragments. Surprisingly this is also true for the first fragment, because generally the packet dissector cannot be sure how large the next protocol headers are, and it cannot stop the dissection process at the fragmentation boundary. You can see this in the Ipv4ProtocolDissector.cc file at lines 32-38. 

Therefore the UDP header is not found inside the packet by the packet filter expression and a nullptr is returned. Finally, the destination port member is searched on the nullptr, which obviously doesn't work. Even if the first fragment would be dissected somehow up to the UDP header, subsequent fragments wouldn't work because they don't even have UDP/IP headers in them.

To summarize, for fragmented IP datagrams the udp.destPort expression cannot be used to identify the packets of the application stream. Instead, you could add a PCP tag to the packet in the application, for example, and use the tag in the packet filter expression to identify the stream.

**Comment by howhangliu** [comment]
Thank you for the reply.

Currently, I just use the name of the application to filter the packet. It is working.

i.e.
```
#client stream identification
*.tactile.bridging.streamIdentifier.identifier.mapping = [{stream: "besteffort", packetFilter: expr(name =~ '*besteffort*')},
                                                         {stream: "video", packetFilter: expr(name =~ '*video*')}]
```

Thanks again for the assistance.

**Comment by Faniche** [comment]
> Thank you for the reply.
> 
> Currently, I just use the name of the application to filter the packet. It is working.
> 
> i.e.
> 
> ```
> #client stream identification
> *.tactile.bridging.streamIdentifier.identifier.mapping = [{stream: "besteffort", packetFilter: expr(name =~ '*besteffort*')},
>                                                          {stream: "video", packetFilter: expr(name =~ '*video*')}]
> ```
> 
> Thanks again for the assistance.

Is there more specification for using pakcetFilter? I just found expr with `port` or `"*"`. Thanks so much
